src/rules/enforce-boolean-naming-prefixes.ts

```
@@ -191,38 +191,84 @@ export const enforceBooleanNamingPrefixes = createRule<Options, MessageIds>({
               rightSide.callee.property.type === AST_NODE_TYPES.Identifier
             ) {
               const methodName = rightSide.callee.property.name;
+              const lowerMethodName = methodName.toLowerCase();
+
+              // Ignore assert*-style methods which often return the input value
+              if (lowerMethodName.startsWith('assert')) {
+                return false;
+              }
 
               // Check if the method name suggests it returns a boolean
               const isBooleanMethod = approvedPrefixes.some((prefix) =>
```

ESLint

[error] Insert `⏎················`

(prettier/prettier)